src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -1214,6 +1217,66 @@ private IArrayInitializerOperation CreateBoundArrayInitializationOperation(Bound
             return new ArrayInitializerOperation(elementValues, _semanticModel, syntax, isImplicit);
         }
 
+        private IOperation CreateBoundCollectionLiteralExpression(BoundCollectionLiteralExpression boundCollectionLiteralExpression)
+        {
+            SyntaxNode syntax = boundCollectionLiteralExpression.Syntax;
+            ITypeSymbol? collectionType = boundCollectionLiteralExpression.GetPublicTypeSymbol();
+            bool isImplicit = boundCollectionLiteralExpression.WasCompilerGenerated;
+            ImmutableArray<IOperation> elementValues = CreateFromArray<BoundExpression, IOperation>(boundCollectionLiteralExpression.Initializers);
+            IOperation collectionCreation;
+            if (collectionType is IArrayTypeSymbol arrayType)
```

Comment:
nit: consider extracting all these branches into a helper method.  then that helper method just returns the IOp for collection creation, and each branch just does an unconditional 'return'.  it makes it easier to understand versus having to see all these potentially mutate local state and then track that flow to the final line.

Reply:
Unfortunately, we `return new NoneOperation()` directly in the `else` so we'd need to check the result of helper method to avoid wrapping in a `CollectionLiteralOperation`. Perhaps we can extract the helper method once the `else` case (the natural type case) is supported.